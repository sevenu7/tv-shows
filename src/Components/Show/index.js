import React from 'react';
import { Col } from 'react-bootstrap';
import { Link } from 'react-router-dom/cjs/react-router-dom.min';

function Show({ image, name, show }) {
    return (
        <Col xs={6} sm={4} md={2} lg={2} xl={2} className="thumbnail">
            <Link to={{ pathname: `/show/${show.id}`, show }}>
                <img src={image} alt={name} style={{ width: '100%' }} />
                <div className="thumbnail-desc">{name}</div>
            </Link>
        </Col>
    );
}

export default Show;
