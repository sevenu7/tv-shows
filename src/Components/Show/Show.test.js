import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Show from "./index";

configure({ adapter: new Adapter() });

let wrapper;
const show = {
  id: 1,
  name: "Under the Dome",
  genres: ["Drama", "Science-Fiction", "Thriller"],
  schedule: { time: "22:00", days: ["Thursday"] },
  rating: { average: 6.5 },
  image: {
    medium:
      "http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg",
  },
};

beforeEach(() => {
  wrapper = shallow(
    <Show show={show} image={show.image.medium} name={show.name} />
  );
});

describe("<Show />", () => {
  it("should match Snapshot with props", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
