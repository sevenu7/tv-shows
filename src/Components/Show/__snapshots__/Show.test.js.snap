// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Show /> should match Snapshot with props 1`] = `
"<Col xs={6} sm={4} md={2} lg={2} xl={2} className=\\"thumbnail\\">
  <ForwardRef to={{...}}>
    <img src=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" alt=\\"Under the Dome\\" style={{...}} />
    <div className=\\"thumbnail-desc\\">
      Under the Dome
    </div>
  </ForwardRef>
</Col>"
`;
